File: xjakt00-adventura/PrikazCoje.java
/*******************************************************************************
 * Tato třída impelmentuje pro hru příkaz 'coje'. Tento příkaz může mít
 * dva parametry - 'batoh' nebo 'tady'. V každém případě hráči ukáže věci
 * (a případně i osoby), které se v batohu/daném prostoru vyskytují.
 *
 * @author    Tomáš Jakubčin (xjakt00)
 * @version   pro školní rok 2010/2011
 */
public class PrikazCoje implements IPrikaz {
    private static final String NAZEV = "coje";
    private HerniPlan plan;
    private Batoh batoh;
    private Hra hra;

    /***************************************************************************
     *  Konstruktor třídy 
     *  
     *  @param  hra     odkaz na hru.
     *  @param  batoh   odkaz na hráčův batoh.
     *  @param  plan    odkaz na herní plán.
     */
    public PrikazCoje (Batoh batoh, HerniPlan plan, Hra hra) {
        this.plan = plan;
        this.batoh = batoh;
        this.hra = hra;
    }

    /***************************************************************************
     * Tato metoda provádí daný příkaz.
     * Kontroluje, zda jsou správně zadáný parametry a v případě, že ne, informuje
     * o tom hráče. Jinak zavolá seznam, pomocí něhož vypíše vše, co se na daném místě
     * nachází.
     */
    public String proved (String... parametry) {
        if (parametry.length != 1) {
            return "Prikaz musi mit prave jeden parametr 'tady' ";
        }
        String kde = parametry[0];
        if (kde.equals("tady")) {
            if (plan.getCas()-1 > 0) {
                plan.setCas(1); 
                return plan.getAktualniProstor().seznamVeciAOsob();
            }
            else {
                plan.konecHry();
                return "";
            }
        }
        if (kde.equals("batoh")) {
            if (plan.getCas()-1 > 0) {
                plan.setCas(1);
                return batoh.seznamVeci();
            }
            else {
                plan.konecHry();
                return "";
            }
        }
        return "Prikaz musi mit tvar: 'coje tady' nebo 'coje batoh' ";
    }

    /**
     *  Metoda vrací název příkazu (slovo které používá hráč pro jeho vyvolání)
     *  
     *  @ return nazev prikazu
     */
    public String getNazev() {
        return NAZEV;
    }

}

File: src/Adventura/PrikazCoJe.cpp
#include "PrikazCoJe.h"

#include "Hra.h"

PrikazCoJe::PrikazCoJe(const std::string & nazev, Hra * hra, HerniPlan * herni_plan, Batoh * batoh)
	: IPrikaz(nazev), odkaz_na_hru(hra), odkaz_na_plan(herni_plan), odkaz_na_batoh(batoh)
{
}

PrikazCoJe::~PrikazCoJe()
{
}

void PrikazCoJe::proved(const std::vector<std::string> & parametry)
{
	if (parametry.size() == 1)
	{
		if (parametry[0] == "tady")
		{
			auto aktualni_prostor = odkaz_na_plan->get_aktualni_prostor();
			aktualni_prostor->vypis_osoby();
			aktualni_prostor->vypis_veci();
			return;
		}
		if (parametry[0] == "batoh")
		{
			auto seznam_veci = odkaz_na_batoh->seznam_veci();
			if (seznam_veci.empty())
				std::cout << "V batohu nemas zadne veci.\n";
			else
			{
				for (const auto & i : seznam_veci)
				{
					std::cout << i.first.jmeno() << " - " << i.second << "x, ";
				}
				std::cout << '\n';
			}
			
			std::cout << "Mas " << odkaz_na_batoh->get_penize() << " korun.\n";
			return;
		}
	}
	
	std::cout << "Prikaz coje musi mit prave jeden parametr.\n";
}

std::string PrikazCoJe::nazev(void) const
{
	return IPrikaz::jmeno_prikazu;
}

std::string PrikazCoJe::pouziti(void)const
{
	return "coje <tady|batoh>";
}